src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/ProfessionalOrganisationRetriever.java

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.clients;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientResponseException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.iacaseapi.domain.UserDetailsProvider;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.UserDetails;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ref.OrganisationEntityResponse;
+
+@Slf4j
+@Service
+public class ProfessionalOrganisationRetriever {
+
+    private static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+
+    private final RestTemplate restTemplate;
+    private final AuthTokenGenerator serviceAuthTokenGenerator;
+    private final UserDetailsProvider userDetailsProvider;
+    private final String refDataApiUrl;
+    private final String refDataApiPath;
+
+    public ProfessionalOrganisationRetriever(RestTemplate restTemplate,
+                                             AuthTokenGenerator serviceAuthTokenGenerator,
+                                             UserDetailsProvider userDetailsProvider,
+                                             @Value("${prof.ref.data.url}") String refDataApiUrl,
+                                             @Value("${prof.ref.data.path.org.organisation}") String refDataApiPath) {
+        this.restTemplate = restTemplate;
+        this.serviceAuthTokenGenerator = serviceAuthTokenGenerator;
+        this.userDetailsProvider = userDetailsProvider;
+        this.refDataApiUrl = refDataApiUrl;
+        this.refDataApiPath = refDataApiPath;
+    }
+
+    public OrganisationEntityResponse retrieve() {
+
+        final String serviceAuthorizationToken = serviceAuthTokenGenerator.generate();
+        final UserDetails userDetails = userDetailsProvider.getUserDetails();
+        final String accessToken = userDetails.getAccessToken();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.set(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
+        headers.set(HttpHeaders.AUTHORIZATION, accessToken);
+        headers.set(SERVICE_AUTHORIZATION, serviceAuthorizationToken);
+
+        HttpEntity<String> requestEntity = new HttpEntity<>(headers);
+
+        OrganisationEntityResponse response;
+
+        log.info("Calling Ref Data endpoint: {}", refDataApiUrl + refDataApiPath);
+        log.info("Authorization: {}", accessToken);
```

Comment:
we have to remove those logs, it might be a part of different ticket, but we should not log tokens

Reply:
removed